### PR TITLE
Created specs for python 2 packages

### PR DESCRIPTION
**Pull request by D1v38om83r**
Swagger spec update for new feature. Python 2 package support in Azure Automation.

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by D1v38om83r** [comment]
@ravbhatnagar can you please review on behalf of the ARM team. I am introducing a new resource here that is practically the same as Azure automation Module (powershell), but for python.

**Comment by D1v38om83r** [comment]
ping!
@ravbhatnagar can you please review on behalf of the ARM team. I am introducing a new resource here that is practically the same as Azure automation Module (powershell), but for python.

File: specification/automation/resource-manager/Microsoft.Automation/stable/2018-06-30/python2package.json [hunk 1]
```diff
@@ -0,0 +1,733 @@
...
+        }
+      },
+      "required": [
+        "properties"
+      ],
+      "description": "The parameters supplied to the create or update module operation."
+    },
+    "ModuleProperties": {
+      "properties": {
+        "isGlobal": {
```
Comment (ravbhatnagar) [inline specification/automation/resource-manager/Microsoft.Automation/stable/2018-06-30/python2package.json]: bools are typically not recommended as they are less descriptive and dont allow for future expansion of states if needed.

File: specification/automation/resource-manager/Microsoft.Automation/stable/2018-06-30/python2package.json [hunk 2]
```diff
@@ -0,0 +1,733 @@
...
+          "type": "string",
+          "format": "date-time",
+          "description": "Gets or sets the last modified time.",
+          "x-nullable": false
+        },
+        "description": {
+          "type": "string",
+          "description": "Gets or sets the description."
+        },
+        "isComposite": {
```
Comment (ravbhatnagar) [inline specification/automation/resource-manager/Microsoft.Automation/stable/2018-06-30/python2package.json]: same comment for bool

File: specification/automation/resource-manager/Microsoft.Automation/stable/2018-06-30/python2package.json [hunk 3]
```diff
@@ -0,0 +1,733 @@
...
+      "description": "The parameters supplied to the update properties."
+    },
+    "ModuleUpdateParameters": {
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/ModuleUpdateProperties",
+          "x-ms-client-flatten": true,
+          "description": "Gets or sets the module update properties."
+        },
+        "name": {
```
Comment (ravbhatnagar) [inline specification/automation/resource-manager/Microsoft.Automation/stable/2018-06-30/python2package.json]: Is name of the resource really updatable?

File: specification/automation/resource-manager/Microsoft.Automation/stable/2018-06-30/python2package.json [hunk 4]
```diff
@@ -0,0 +1,733 @@
...
+          "type": "string",
+          "description": "Gets or sets name of the resource."
+        },
+        "location": {
```
Comment (ravbhatnagar) [inline specification/automation/resource-manager/Microsoft.Automation/stable/2018-06-30/python2package.json]: location of azure resources is not updatbale.

File: specification/automation/resource-manager/Microsoft.Automation/stable/2018-06-30/python2package.json [hunk 5]
```diff
@@ -0,0 +1,733 @@
...
+      "description": "The parameters supplied to the create or update module properties."
+    },
+    "ModuleCreateOrUpdateParameters": {
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/ModuleCreateOrUpdateProperties",
+          "x-ms-client-flatten": true,
+          "description": "Gets or sets the module create properties."
+        },
+        "name": {
```
Comment (ravbhatnagar) [inline specification/automation/resource-manager/Microsoft.Automation/stable/2018-06-30/python2package.json]: name comes from the URL. It should be readonly in the request body

File: specification/automation/resource-manager/Microsoft.Automation/stable/2018-06-30/python2package.json [hunk 6]
```diff
@@ -0,0 +1,733 @@
...
+          "$ref": "#/definitions/ContentLink",
+          "description": "Gets or sets the module content link."
+        }
+      },
+      "required": [
+        "contentLink"
+      ],
+      "description": "The parameters supplied to the create or update module properties."
+    },
+    "ModuleCreateOrUpdateParameters": {
```
Comment (ravbhatnagar) [inline specification/automation/resource-manager/Microsoft.Automation/stable/2018-06-30/python2package.json]: why is the same model no used for request and response? It helps in GET-PUT pipeline and makes it simpler to consume

File: specification/automation/resource-manager/Microsoft.Automation/stable/2018-06-30/python2package.json (hunk 1 above)
Comment (D1v38om83r) [reply specification/automation/resource-manager/Microsoft.Automation/stable/2018-06-30/python2package.json]: The property isGlobal is set in out data model as a boolean everywhere. Its not expected to change ever. The specs for Python 2 packages are practically the same a Powershell modules for consistency, except for paths.

File: specification/automation/resource-manager/Microsoft.Automation/stable/2018-06-30/python2package.json (hunk 2 above)
Comment (D1v38om83r) [reply specification/automation/resource-manager/Microsoft.Automation/stable/2018-06-30/python2package.json]: isComposite is never expected to change. The specs for Python 2 packages are practically the same a Powershell modules for consistency, except for paths.

File: specification/automation/resource-manager/Microsoft.Automation/stable/2018-06-30/python2package.json (hunk 3 above)
Comment (D1v38om83r) [reply specification/automation/resource-manager/Microsoft.Automation/stable/2018-06-30/python2package.json]: no, only the tag is updatable. Left in there for consistency with PowerShell modules.

File: specification/automation/resource-manager/Microsoft.Automation/stable/2018-06-30/python2package.json (hunk 4 above)
Comment (D1v38om83r) [reply specification/automation/resource-manager/Microsoft.Automation/stable/2018-06-30/python2package.json]: yes, that is correct. Left in there for consistency with PowerShell modules.

**Comment by jhendrixMSFT** [comment]
Please be sure to review the fixes made to the powershell equivalent as there were some bugs in that swagger which made the "GetContent" API not work, see the [history](https://github.com/Azure/azure-rest-api-specs/commits/master/specification/automation/resource-manager/Microsoft.Automation).

File: specification/automation/resource-manager/Microsoft.Automation/stable/2018-06-30/python2package.json (hunk 3 above)
Comment (ravbhatnagar) [reply specification/automation/resource-manager/Microsoft.Automation/stable/2018-06-30/python2package.json]: Consistency at the expense of something which doesnt work is not recommended. Please fix.

File: specification/automation/resource-manager/Microsoft.Automation/stable/2018-06-30/python2package.json (hunk 4 above)
Comment (ravbhatnagar) [reply specification/automation/resource-manager/Microsoft.Automation/stable/2018-06-30/python2package.json]: same comment as above. lets fix this. And even fix powershell in a new api-version.

**Comment by jhendrixMSFT** [comment]
@D1v38om83r any update on ARM feedback?

File: specification/automation/resource-manager/Microsoft.Automation/stable/2018-06-30/python2package.json (hunk 4 above)
Comment (D1v38om83r) [reply specification/automation/resource-manager/Microsoft.Automation/stable/2018-06-30/python2package.json]: I have removed the parameters which have no effect.

File: specification/automation/resource-manager/Microsoft.Automation/stable/2018-06-30/python2package.json (hunk 5 above)
Comment (D1v38om83r) [reply specification/automation/resource-manager/Microsoft.Automation/stable/2018-06-30/python2package.json]: Parameter name has been dropped

**Comment by jhendrixMSFT** [comment]
@D1v38om83r there is a merge conflict, can you please rebase your changes on top of master?

**Comment by jhendrixMSFT** [comment]
@D1v38om83r ping

**Comment by D1v38om83r** [comment]
@jhendrixMSFT the merge conflicts have been resolved

**Comment by jhendrixMSFT** [comment]
@D1v38om83r thanks!  Where are we with the ARM review?

**Comment by jhendrixMSFT** [comment]
@D1v38om83r also has this service been deployed?

**Comment by jhendrixMSFT** [comment]
Looking at the ARM feedback I think the only remaining issue is why is the same model not used for request and response.  Can this be fixed?

**Comment by D1v38om83r** [comment]
The request and response are too dissimilar, there is next to no overlap. It doesn't make sense to combine them.

**Comment by jhendrixMSFT** [comment]
Thanks for the explanation this makes sense.  Will merge EOD unless @ravbhatnagar has any objections.

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**